apis/v1beta1/config_test.go

```diff
@@ -277,7 +277,7 @@ func TestConfigToMetricsPort(t *testing.T) {
 	} {
 		t.Run(tt.desc, func(t *testing.T) {
 			// these are acceptable failures, we return to the collector's default metric port
-			port, err := tt.config.MetricsPort()
+			_, port, err := tt.config.MetricsEndpoint()
```

Comment:
could we change this test and assert on the endpoint as well? There could be also another test with setting the address to a custom one

Reply:
done.